triedb/firewood/database.go

```diff
@@ -375,10 +375,15 @@ func (db *Database) Close() error {
 	db.proposalLock.Lock()
 	defer db.proposalLock.Unlock()
 
-	// We don't need to explicitly dereference the proposals, since they will be cleaned up
-	// within the firewood close method.
+	// clear the map before calling `dereference` to prevent extra work on close
 	db.proposalMap = nil
```

Comment:
is `clear()` not preferred here for the map? The new comment is unclear to me

Reply:
on reflection, setting this to nil is preferred for _this case_ because I'm specifically checking for nil instead of len == 0. Checking for len==0 would work but feels less correct since nil indicates an invalid state (we always `make` in the New function).